### PR TITLE
Add filterrific

### PR DESCRIPTION
**Pull request by latazzajones**
What this pr does: 
- Replaces the search functionality on the Friends index page with filterrific. 
- Removes the no longer used app/helpers/friends_helper.rb
- Adds all the scopes needed for the new filterrific search. 
- Pulls the friends search out into a partial. 

What's left to do: 
- Replace all search functionality with filterrific
- Add Readme directions for adding new search fields (I'll try to do that this week but I don't want this work to get held up by that)
- There is some wierdness with the way that the textacular gem matches against full text. So, if you search with "Ro" it may return "Rob" but not "Roberta" since it looks for words that match by %. To address this, I'd like to replace textacular with either the [pg-search gem](https://github.com/Casecommons/pg_search) or something similar. 
- Maybe have someone more savey in front end stuff make it look prettier :-) Please let me know if you want me to make some little changes now.

screen shot 2018-12-09 at 11 52 01 am

### CODE REVIEW

**Comment by CZagrobelny** [comment]
@latazzajones this is so awesome!!! re: little front end changes, I think first_name, last_name, a_number all on the same row, if that looks okay?

**Comment by latazzajones** [comment]
@CZagrobelny ya - I tried for about 40mins and gave up. I couldn't get it to look right. I'll give it one more shot sometime this week.

**Comment by latazzajones** [comment]
@CZagrobelny Here's what it looks like now: 

screen shot 2018-12-12 at 5 49 54 pm

For the life of me I can't figure out how to get the "With 1 year" text to line up under the form fields above it and it's driving me crazy :-/ ideas?

**Comment by latazzajones** [comment]
@CZagrobelny here's the latest iteration 👏 thanks to @jgaskins!! 
49947284-d83e4100-febe-11e8-83ac-88b9af1bded8
49947321-f1df8880-febe-11e8-9200-d8e52faacdfa

**Comment by CZagrobelny** [comment]
LOVE IT!  Thanks @latazzajones and @jgaskins !

File: spec/models/friend_spec.rb [hunk 1]
```diff
@@ -127,35 +127,109 @@
...
-        outside_boundaries = [
-          create(:friend, date_of_entry: earliest_date_of_entry - 1.week),
-          create(:friend, date_of_entry: earliest_date_of_entry - 1.day),
-          create(:friend, date_of_entry: latest_date_of_entry + 1.day),
-          create(:friend, date_of_entry: latest_date_of_entry + 1.week)
-        ]
-
-        friends = Friend.asylum_deadlines_ending(deadlines_ending_floor, deadlines_ending_ceiling)
-        expect(friends).to eq(within_boundaries)
+  describe '#filter_a_number' do
```
Comment (CZagrobelny) [inline spec/models/friend_spec.rb]: These new specs are so awesome, makes it so clear how everything works!

**Pull request merged**